src/git.rs

```diff
@@ -227,6 +231,8 @@ impl Repository {
             ref_status = Err("update_reference callback was not called".into());
         }
 
+        println!("commit_and_push: Finished with \"{}\"", msg);
```

Comment:
I think this final step should still be extracted into a wrapper function.  If any of the early returns or `?`s produce an error, then nothing will be logged.  I think a wrapper like the following would be an easy way to address that:

```rust
fn commit_and_push(&self, msg: &str, modified_file: &Path) -> Result<(), PerformError> {
    self.commit_and_push_inner(msg, modified_file)
        .map(|_| println!("commit_and_push: Finished with \"{}\"", msg))
        .map_err(|err| {
            println!("commit_and_push: Finished with \"{}\"", err);
            err
        })
}
```

Reply:
I think I've settled on a good solution very similar to this, but with the initial println! also in the wrapper method and slightly different naming.